cmd/osquery-perf/agent.go

```diff
@@ -2150,6 +2150,55 @@ func (a *agent) submitLogs(results []resultLog) error {
 	return nil
 }
 
+func runAppleIDeviceMDMLoop(i int, stats *Stats, model string, serverURL string, mdmSCEPChallenge string, mdmCheckInInterval time.Duration) {
+	udid := mdmtest.RandUDID()
+
+	mdmClient := mdmtest.NewTestMDMClientAppleDirect(mdmtest.AppleEnrollInfo{
+		SCEPChallenge: mdmSCEPChallenge,
+		SCEPURL:       serverURL + apple_mdm.SCEPPath,
+		MDMURL:        serverURL + apple_mdm.MDMPath,
+	}, model)
+	mdmClient.UUID = udid
+	mdmClient.SerialNumber = mdmtest.RandSerialNumber()
+	deviceName := fmt.Sprintf("%s-%d", model, i)
+	productName := model
+
+	if err := mdmClient.Enroll(); err != nil {
+		log.Printf("%s MDM enroll failed: %s", model, err)
+		stats.IncrementMDMErrors()
+		return
+	}
+
+	stats.IncrementMDMEnrollments()
+
+	mdmCheckInTicker := time.Tick(mdmCheckInInterval)
+
+	for range mdmCheckInTicker {
+		mdmCommandPayload, err := mdmClient.Idle()
+		if err != nil {
+			log.Printf("MDM Idle request failed: %s: %s", model, err)
+			stats.IncrementMDMErrors()
+			continue
+		}
+		stats.IncrementMDMSessions()
+
+	INNER_FOR_LOOP:
+		for mdmCommandPayload != nil {
+			stats.IncrementMDMCommandsReceived()
+			if mdmCommandPayload.Command.RequestType == "DeviceInformation" {
+				mdmCommandPayload, err = mdmClient.AcknowledgeDeviceInformation(udid, mdmCommandPayload.CommandUUID, deviceName, productName)
+			} else {
+				mdmCommandPayload, err = mdmClient.Acknowledge(mdmCommandPayload.CommandUUID)
+			}
+			if err != nil {
+				log.Printf("MDM Acknowledge request failed: %s: %s", model, err)
+				stats.IncrementMDMErrors()
+				break INNER_FOR_LOOP
```

Comment:
Nit, but the label is not necessary - that's the `for` loop that this would `break` out of.